NitroxServer-Subnautica/Serialization/ResourceAssets/MonoBehaviourParsers/DefaultMonoBehaviourParser.cs

```diff
@@ -0,0 +1,97 @@
+                    for (int a = 0; a < placeholders; a++)
+                    {
+                        resourcesFile.reader.ReadInt32();//FileId
+                        ulong pathId = (ulong)resourcesFile.reader.ReadInt64();//PathId
+                        AssetFileInfoEx prefabPlaceholder = resourcesFileTable.getAssetInfo(pathId);
+                        ulong placeholderPos = resourcesFile.reader.Position;
+                        resourcesFile.reader.Position = prefabPlaceholder.absoluteFilePos;
+                        resourcesFile.reader.ReadInt32();
+                        ulong gameObjectPath = (ulong)resourcesFile.reader.ReadInt64();
+                        AssetFileInfoEx transformAsset = ResourceAssetsHelper.FindTransform(instance, gameObjectPath);
+
+                        ulong curPos = resourcesFile.reader.Position;
+                        resourcesFile.reader.Position = transformAsset.absoluteFilePos + 12;
+
+                        Quaternion rotation = new Quaternion(
+                            afr.ReadSingle(), // Quaternion X
+                            afr.ReadSingle(), // Quaternion Y
+                            afr.ReadSingle(), // Quaternion Z
+                            afr.ReadSingle()); // Quaternion W
+
+                        Vector3 position = new Vector3(
+                            afr.ReadSingle(), // Position X
+                            afr.ReadSingle(), // Position Y
+                            afr.ReadSingle()); // Position Z
+
+                        Vector3 scale = new Vector3(
+                            afr.ReadSingle(), // Scale X
+                            afr.ReadSingle(), // Scale Y
+                            afr.ReadSingle()); // Scale Z
+
+                        resourcesFile.reader.Position = curPos;
+                        resourcesFile.reader.Position += 16;
+
+                        resourcesFile.reader.ReadCountStringInt32();//Empty...
+                        string prefabString = resourcesFile.reader.ReadCountStringInt32();
+
+                        prefabs.Add(new UwePrefab(prefabString, position, rotation, scale));
+                        resourcesFile.reader.Position = placeholderPos; // Reset position
+                    }
+
+                    AssetFileInfoEx prefabIdentifierAsset = ResourceAssetsHelper.FindComponentOfType<PrefabIdentifier>(instance, (ulong)gameObjectPathId);
+                    resourcesFile.reader.Position = prefabIdentifierAsset.absoluteFilePos + 28;
+
+                    resourcesFile.reader.ReadCountStringInt32(); // Empty name
+                    string classId = resourcesFile.reader.ReadCountStringInt32(); // ClassId
+
+                    List<UwePrefab> val;
+                    if (!SubnauticaUwePrefabFactory.PrefabsByClassId.TryGetValue(classId, out val)) // should never be false but just incase
+                    {
+                        SubnauticaUwePrefabFactory.PrefabsByClassId.Add(classId, prefabs);
```

Comment:
I think if you use the [] accessor it won't throw an exception if it already exists... that way you can remove the if check.

```
 SubnauticaUwePrefabFactory.PrefabsByClassId[classId] = prefabs;
```